aten/src/ATen/native/native_functions.yaml

```diff
@@ -1153,6 +1153,13 @@
     CPU: dot
     CUDA: dot_cuda
 
+- func: vdot(Tensor self, Tensor tensor) -> Tensor
+  use_c10_dispatcher: full
+  variants: function
+  dispatch:
+    CPU: vdot
+    CUDA: vdot_cuda
+
 - func: dot.out(Tensor self, Tensor tensor, *, Tensor(a!) out) -> Tensor(a!)
```

Comment:
Better put vdot after all the dot functions (and add vdot.out?)

Reply:
Would out variant be required here? As output really is a single element tensor. Also numpy doesn't have `out` in its [signature](https://numpy.org/doc/stable/reference/generated/numpy.vdot.html)

However as mentioned `dot` does have a out variant.

Reply:
It's very easy to add and we should be consistent with `torch.dot`.

Reply:
Sure. Will do that.